src/components/cursor.js

```diff
@@ -33,62 +39,140 @@ module.exports.Component = registerComponent('cursor', {
   dependencies: ['raycaster'],
 
   schema: {
+    downEvents: {default: []},
     fuse: {default: utils.device.isMobile()},
-    fuseTimeout: {default: 1500, min: 0}
+    fuseTimeout: {default: 1500, min: 0},
+    projectionEnabled: {default: false},
+    projectionMixin: {default: ''},
+    laserEnabled: {default: false},
+    upEvents: {default: []}
   },
 
   init: function () {
+    var data = this.data;
     var el = this.el;
-    var canvas = el.sceneEl.canvas;
+    var raycasterData;
+
     this.fuseTimeout = undefined;
-    this.mouseDownEl = null;
+    this.cursorDownEl = null;
     this.intersection = null;
     this.intersectedEl = null;
 
-    // Wait for canvas to load.
-    if (!canvas) {
-      el.sceneEl.addEventListener('render-target-loaded', bind(this.init, this));
-      return;
-    }
-
     // Bind methods.
-    this.onMouseDown = bind(this.onMouseDown, this);
-    this.onMouseUp = bind(this.onMouseUp, this);
+    this.onCursorDown = bind(this.onCursorDown, this);
+    this.onCursorUp = bind(this.onCursorUp, this);
     this.onIntersection = bind(this.onIntersection, this);
     this.onIntersectionCleared = bind(this.onIntersectionCleared, this);
 
-    // Attach event listeners.
-    canvas.addEventListener('mousedown', this.onMouseDown);
-    canvas.addEventListener('mouseup', this.onMouseUp);
-    el.addEventListener('raycaster-intersection', this.onIntersection);
-    el.addEventListener('raycaster-intersection-cleared', this.onIntersectionCleared);
+    if (data.projectionEnabled) {
+      this.injectProjectionEl();
+      // If projection enabled, speed up interval so mesh position updates are smooth.
+      raycasterData = el.getDOMAttribute('raycaster');
+      if (!raycasterData || !('interval' in raycasterData)) {
+        el.setAttribute('raycaster', 'interval', 15);
+      }
```

Comment:
This is for projecting a mesh on the intersection point, not the laser. Without it, it gets very choppy.